### PR TITLE
ITX transactor provider

### PR DESCRIPTION
**Pull request by mpetrun5**
- moves sign and send transactor to separate package to allow for more transactors
- add itx transactor with gsn forwarder implmementation
- allow for store to be more generic and separate it into blockstore and noncestore

## Related Issue Or Context

Closes: #68  

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [x] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

**Comment by mpetrun5** [comment]
This is pretty much ready for review. Still we need the PR on contracts to be merged first to have everything testned manually, but other than that the code should work.

**Comment by mpetrun5** [comment]
Successfully tested with new contracts.

File: chains/evm/calls/transactor/signAndSend/signAndSend.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	tx, err := t.TxFabric(n.Uint64(), to, opts.Value, opts.GasLimit, gp, data)
+	if err != nil {
+		return &common.Hash{}, err
+	}
+
+	h, err := t.client.SignAndSendTransaction(context.TODO(), tx)
+	if err != nil {
+		log.Error().Err(err).Msg("SIGN_AND_SEND")
```
Comment (MakMuftic) [inline chains/evm/calls/transactor/signAndSend/signAndSend.go:61]: Is there a reason for this error message to be formated like this?

**Review by MakMuftic — CHANGES_REQUESTED** [review]

**Review by vezenovm — APPROVED** [review]
LGTM, just Mak's couple comments. Also, in the original PR (https://github.com/ChainSafe/ChainBridge/pull/653) there is an implementation of a Gnosis Forwarder which could be very useful for relayers. For now that seems out of scope though and is something that can be discussed and added in a separate PR.

File: chains/evm/calls/transactor/signAndSend/signAndSend.go (hunk 1 above)
Comment (mpetrun5) [reply chains/evm/calls/transactor/signAndSend/signAndSend.go:61]: Copy pasted from old implementation and didn't think too much about it

File: chains/evm/calls/contracts/forwarder/forwarder.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	res, err := c.CallContract("getNonce", from)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := abi.ConvertType(res[0], new(big.Int)).(*big.Int)
+	return nonce, nil
+}
+
+func (c *ForwarderContract) ExecuteData(
```
[the remark this answers is not in the document]
Comment (mpetrun5) [reply chains/evm/calls/contracts/forwarder/forwarder.go:49]: It is returning data for the 'execute' function. Should I call it something like 'PackExecute'? Haven't really come up with anything clear.

File: chains/evm/calls/transactor/itx/itx.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	txID := crypto.Keccak256Hash(packed)
+	msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(txID), string(txID.Bytes()))
+	hash := crypto.Keccak256Hash([]byte(msg))
+	sig, err := crypto.Sign(hash.Bytes(), itx.kp.PrivateKey())
+	if err != nil {
+		return nil, nil
```
Comment (tcar121293) [inline chains/evm/calls/transactor/itx/itx.go:135]: You should probably return the error?

File: chains/evm/calls/transactor/signAndSend/signAndSend.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	defer t.client.UnlockNonce()
+	t.client.LockNonce()
+	n, err := t.client.UnsafeNonce()
+	if err != nil {
+		return &common.Hash{}, nil
+	}
+
+	err = transactor.MergeTransactionOptions(&opts, &DefaultTransactionOptions)
+	if err != nil {
+		return &common.Hash{}, nil
```
Comment (tcar121293) [inline chains/evm/calls/transactor/signAndSend/signAndSend.go:43]: Return err?

File: chains/evm/calls/transactor/signAndSend/signAndSend.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		client:         client,
+	}
+}
+
+func (t *signAndSendTransactor) Transact(to *common.Address, data []byte, opts transactor.TransactOptions) (*common.Hash, error) {
+	defer t.client.UnlockNonce()
+	t.client.LockNonce()
+	n, err := t.client.UnsafeNonce()
+	if err != nil {
+		return &common.Hash{}, nil
```
Comment (tcar121293) [inline chains/evm/calls/transactor/signAndSend/signAndSend.go:38]: Return err?

**Review by MakMuftic — APPROVED** [review]

**Review by tcar121293 — APPROVED** [review]

**Pull request merged**